odl/tomo/geometry/geometry.py

```diff
@@ -155,83 +173,189 @@ def translation(self):
+            If ``mparam`` is a single parameter, a single vector is
+            returned, otherwise a stack of vectors along axis 0.
         """
         raise NotImplementedError('abstract method')
 
-    def rotation_matrix(self, mpar):
-        """Return the rotation matrix from initial state to state at ``mpar``.
+    def rotation_matrix(self, mparam):
+        """Return the rotation matrix to the system state at ``mparam``.
 
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter for which to calculate the rotation matrix.
+        mparam : `array-like`
+            Motion parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
 
         Returns
         -------
-        rot : `numpy.ndarray`, shape (`ndim`, `ndim`)
-            The rotation matrix mapping vectors at the initial state
-            to the ones in the state defined by ``mpar``. The rotation
-            is extrinsic, i.e., defined in the fixed ("world") coordinate
-            system.
+        rot : `numpy.ndarray`, shape (ndim, ndim) or (num_mparams, ndim, ndim)
+            The rotation matrix (or matrices) mapping vectors at the
+            initial state to the ones in the state defined by ``mparam``.
+            The rotation is extrinsic, i.e., defined in the "world"
+            coordinate system.
+            If ``mparam`` is a single parameter, a single matrix is
+            returned, otherwise a stack of matrices along axis 0.
         """
         raise NotImplementedError('abstract method')
 
-    def det_to_src(self, mpar, dpar, normalized=True):
+    def det_to_src(self, mparam, dparam, normalized=True):
         """Vector pointing from a detector location to the source.
 
-        A function of the motion and detector parameters.
-
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter at which to evaluate.
-        dpar : `det_params` element
-            Detector parameter at which to evaluate.
+        mparam : `motion_params` element or `array-like`
+            Motion parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
+        dparam : `det_params` element or `array-like`
+            Detector parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
         normalized : bool, optional
-            If ``True``, return a normalized (unit) vector.
+            If ``True``, normalize the resulting vector(s) to unit length.
 
         Returns
         -------
-        vec : `numpy.ndarray`, shape (`ndim`,)
-            (Unit) vector pointing from the detector to the source
+        vec : `numpy.ndarray`
+            (Unit) vector(s) pointing from the detector to the source.
+            If both ``mparam`` and ``dparam`` are single parameters, a single
+            vector is returned, otherwise a stack of vectors.
```

Comment:
Not quite, only the cases described in the doc, not something like a 2d array of parameters. But with some minor tweaks in the code it should be simple to allow that, too.